stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/list/ChannelListFragment.kt

```diff
@@ -27,7 +29,15 @@ import io.getstream.chat.ui.sample.feature.home.HomeFragmentDirections
 
 class ChannelListFragment : Fragment() {
 
-    private val viewModel: ChannelListViewModel by viewModels { ChannelListViewModelFactory() }
+    private val viewModel: ChannelListViewModel by viewModels {
+        ChannelListViewModelFactory(
+            filter = Filters.and(
+                Filters.eq("type", "messaging"),
+                Filters.`in`("members", listOf(ChatClient.instance().getCurrentUser()?.id ?: "")),
```

Comment:
Setting the user is an async operation and we don't wait until it's finished. There might be a case when this view model is created before the user id set and we'll display wrong channels